coordinator/src/orderbook/db/orders.rs

```diff
@@ -410,6 +410,22 @@ pub fn delete(conn: &mut PgConnection, id: Uuid) -> QueryResult<OrderbookOrder>
     set_order_state(conn, id, commons::OrderState::Deleted)
 }
 
+pub fn set_order_state_partially_taken(
+    conn: &mut PgConnection,
+    id: Uuid,
+    quantity: Decimal,
+) -> QueryResult<OrderbookOrder> {
+    let order: Order = diesel::update(orders::table)
+        .filter(orders::order_id.eq(id))
+        .set((
+            orders::order_state.eq(OrderState::Taken),
+            orders::quantity.eq(quantity.to_f32().expect("to fit")),
```

Comment:
🔔 
I think we shouldn't change the quantity. This will make it hard to do any type of analysis on orders in the future. Instead, we should add an additional field which indicates how much has been used already, e.g. `used_quantity`.
This can all happen on the db layer if we don't care about this at the moment, i.e. when loading the order from the orderbook we set the in-memory `quantity`=`quantity`-`used_quantity`

Reply:
This is only done if the user deletes his order. If he deletes the order which has already been partially matched. the order quantity is updated to the matched amount and the state is set to `Taken`. I found this to be the most correct way of handling a delete order request of an already partially matched order.

Would you still add an optional `used_quantity` field which is only set when the user deletes his order? and what should be the state of the order `Taken` or `Deleted`?

Reply:
Ah ok, I think I misunderstood this part then.